examples/json/hello/func.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/fnproject/fdk-go"
+)
+
+func main() {
+	fdk.Handle(fdk.HandlerFunc(myHandler))
+}
+
+func myHandler(ctx context.Context, in io.Reader, out io.Writer) {
+	var person struct {
+		Name string `json:"name"`
+	}
+	json.NewDecoder(in).Decode(&person)
+
+	if person.Name == "" {
+		person.Name = "world"
+	}
+
+	out.Write([]byte(fmt.Sprintf("Hello %s!\n", person.Name)))
```

Comment:
don't we need to return json here?

Reply:
i see that the handler does this, so this example is just slightly confusing in that the incoming path decodes a json body but writes back plain text. i understand the protocol underneath is simply an additional layer of json that lets the user write whatever they want in and out, I think this would be slightly less confusing maybe if all plaintext or all json instead of half and half.

Reply:
no, developers should just write response body using this `.Write`

Reply:
it's just a writer, there is an http example that just does too `json.NewEncoder(out).Encode(&myThing)` -- it just means that even with json protocol, http response to the client will be a json body (many people will want this). at one point we wanted to make all examples JSON. in any event, either plain text for everything or json for everything in the example. the mixed I find really confusing.